lib/spack/llnl/util/filesystem.py

```diff
@@ -113,6 +113,69 @@ def path_contains_subdirectory(path, root):
     return norm_path.startswith(norm_root)
 
 
+@memoized
+def file_command(*args):
+    """Creates entry point to `file` system command with provided arguments"""
+    try:
+        file_cmd = which("file", required=True)
+    except CommandNotFoundError as e:
+        if is_windows:
+            raise CommandNotFoundError("`file` utility is not available on Windows")
+        else:
+            raise e
+    for arg in args:
+        file_cmd.add_default_arg(arg)
+    return file_cmd
+
+
+@memoized
+def _get_mime_type():
+    """Generate method to call `file` system command to aquire mime type
+    for a specified path
+    """
+    return file_command("-b", "-h", "--mime-type")
+
+
+@memoized
+def _get_mime_type_compressed():
+    """Same as _get_mime_type but attempts to check for
+    compression first
+    """
+    mime_uncompressed = _get_mime_type()
+    mime_uncompressed.add_default_arg("-Z")
+    return mime_uncompressed
+
+
+def mime_type(filename):
+    """Returns the mime type and subtype of a file.
+
+    Args:
+        filename: file to be analyzed
+
+    Returns:
+        Tuple containing the MIME type and subtype
+    """
+    output = _get_mime_type()(filename, output=str, error=str).strip()
+    tty.debug("==> " + output)
+    type, _, subtype = output.partition("/")
+    return type, subtype
+
+
+def compressed_mime_type(filename):
```

Comment:
(question) is this used anywhere? It looks like `mime_type` moved from `relocate` so that this could be defined (and presumably used by modules that should not otherwise care about `relocate.py`). Moving these functions here is fine (although it's worth knowing whether doing so is essential to this PR).

Reply:
the `mime_type` work was originally going to be our approach before further investigation revealed the magic number parsing was a superieior approach. This is leftover from my initial pass implementing `file` based type parsing that we abandoned. I left it in as it added (ostensibly useful) components and I felt that `relocate` was perhaps not the most logical/visible place for this to live. It is not essential to the PR.